src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -138,18 +81,21 @@ bool CKernelRidgeRegression::train_machine(CFeatures *data)
 	}
 	ASSERT(kernel && kernel->has_features())
 
-	switch (m_train_func)
+	if (m_labels->get_num_labels() != kernel->get_num_vec_rhs())
```

Comment:
adding this check here and not in constructor/setter/base class because
* in constructor or base class labels or kernel may not be defined.
* setters like `set_kernel()` and `set_labels()` can be called in any order so one of these variables can be NULL.
* `train_machine()` does all the checking when it's required.